dissect/target/plugins/os/unix/iptables.py

```diff
@@ -0,0 +1,146 @@
+import re
+from datetime import datetime
+from pathlib import Path
+from typing import Iterator
+
+from dissect.target.helpers.record import TargetRecordDescriptor
+from dissect.target.plugin import Plugin, export
+from dissect.target.target import Target
+
+PATTERN_IPTABLES_SAVE_GENERATED = re.compile(r"# Generated by (ip6?tables-save) v([.\d]+) on (.+)")
+
+PATTERN_IPTABLES_SAVE_POLICY = re.compile(
+    r":(?P<chain>[\w-]+) (?P<policy>[\w-]+|-) \[(?P<packet_count>\d+):(?P<byte_count>\d+)]"
+)
+
+PATTERN_IPTABLES_SAVE_RULE = re.compile(
+    r"(?:\[(?P<packet_count>\d+):(?P<byte_count>\d+)] )?(?P<rule>-A (?P<chain>[\w-]+).+)"
+)
+
+IptablesSaveRecord = TargetRecordDescriptor(
+    "application/firewall/iptables/save",
+    [
+        ("datetime", "ts"),
+        ("string", "program"),
+        ("string", "version"),
+        ("string", "table"),
+        ("string", "chain"),
+        ("string", "type"),
+        ("string", "rule"),
+        ("varint", "packet_count"),
+        ("varint", "byte_count"),
+        ("path", "source"),
+    ],
+)
+
+
+class IptablesSavePlugin(Plugin):
+    """Parser for iptables-save (and ip6tables-save) rules.
+
+    As iptables rules are not stored on disk by default, users
+    that want persistent rules need to store them somewhere and
+    reload them on boot. iptables provides tools to save and reload
+    rules (iptables-save and iptables-restore). These tools
+    do not have a default path to look for rules, however there
+    are multiple commonly used paths.
+
+    References:
+        - https://git.netfilter.org/iptables/
+    """
+
+    COMMON_SAVE_PATHS = (
+        # IPv4
+        "/etc/iptables/rules.v4",
+        "/etc/sysconfig/iptables",
+        "/etc/iptables.rules",
+        "/etc/iptablesRule.v4",
+        "/etc/network/iptables.rules",
+        # IPv6
+        "/etc/iptables/rules.v6",
+        "/etc/iptablesRule.v6",
+        "/etc/sysconfig/ip6tables",
+    )
+
+    LOG_TIME_FORMAT = "%a %b  %d %H:%M:%S %Y"
+
+    def __init__(self, target: Target):
+        super().__init__(target)
+        self._rule_files = list(self._get_rule_files())
+
+    def check_compatible(self) -> bool:
+        return len(self._rule_files) > 0
+
+    def _get_rule_files(self) -> Iterator[Path]:
+        """Yield the paths of iptables-save output files."""
+        for rule_path in self.COMMON_SAVE_PATHS:
+            rule_path = self.target.fs.path(rule_path)
+
+            if rule_path.exists():
+                with rule_path.open("r") as h_rule:
+                    first_line = h_rule.readline()
+
+                if PATTERN_IPTABLES_SAVE_GENERATED.match(first_line):
+                    yield rule_path
+
+    @export(record=IptablesSaveRecord)
+    def iptables(self) -> Iterator[IptablesSaveRecord]:
```

Comment:
Since the file locations are not set in stone, maybe we should offer the option --file to allow the user to point to a specific file instead?

Reply:
For now - in our opinion - the `COMMON_SAVE_PATHS` is sufficient.

Reply:
Ok.